src/Console/Application.php

```diff
@@ -146,35 +155,33 @@ protected function getContainer()
         // Load cli options:
         $input = new ArgvInput();
         $configPath = $input->getParameterOption(['--config', '-c'], $this->getDefaultConfigPath());
+        $output = new ConsoleOutput();
 
         // Build the service container:
         $this->container = ContainerFactory::buildFromConfiguration($configPath);
+        $this->container->set('console.input', $input);
+        $this->container->set('console.output', $output);
 
         return $this->container;
     }
 
     /**
-     * Configure IO of GrumPHP objects
-     *
-     * @param InputInterface  $input
-     * @param OutputInterface $output
+     * {@inheritdoc}
      */
     protected function configureIO(InputInterface $input, OutputInterface $output)
     {
-        parent::configureIO($input, $output);
+        /** @var ConsoleIO $io */
+        $io = $this->container->get('grumphp.io.console');
 
-        $container = $this->getContainer();
+        $input = $io->getInput();
+        $output = $io->getOutput();
 
-        // Register the console input and output to the container
-        $container->set('console.input', $input);
-        $container->set('console.output', $output);
+        parent::configureIO($input, $output);
```

Comment:
This won't work since you now configure the input and output from the container and not the ones that are available in the commands. This means the arguments and options won't work in the commands. As you can see, the configureIO() does not return anything and should just configure the arguments that are being passed to the method.